frontend/src/lib/components/PropertyFilters/PropertyFilters.js

```diff
@@ -1,88 +1,26 @@
-import React, { Component } from 'react'
-import PropTypes from 'prop-types'
-import api from '../../api'
+import React from 'react'
 import { PropertyFilter } from './PropertyFilter'
 import { Button } from 'antd'
+import { useValues, useActions } from 'kea'
+import { propertyFilterLogic } from './propertyFilterLogic'
 
-export class PropertyFilters extends Component {
-    constructor(props) {
-        super(props)
+export function PropertyFilters({ endpoint, propertyFilters, className, style, onChange }) {
+    const { filters } = useValues(propertyFilterLogic({ propertyFilters, endpoint, onChange }))
+    const { newFilter } = useActions(propertyFilterLogic({ onChange }))
```

Comment:
I'm still on the edge regarding passing so many props. Whenever you pass a new set of props, the old ones get overridden... and I see `propertyFilterLogic` being called with props 4 times. 3 times of which with one prop (`onChange`) and the fourth time with 3 props (`propertyFilters`, `endpoint`, `onChange`). This might lead to a situation where the state in the logic is inconsistent with what you would expect.

Reply:
Hm struggling to work out how to fix this. Having a "setInitialPropertyFilters" action with useEffect?

Reply:
If those props are used just for setting defaults (I checked and it seems to be the case), it should be fine. Just if in some listener you ask for `props.endpoint`, it might not exist anymore as `useAction` with just the onChange prop was called after.

Reply:
Coming back to this, I see that you're calling `setInitial` from `afterMount` to set the default value of `filters`. This is fine ... but there's one other (cleaner?) way you can do it.

You can also use props directly to set defaults. Looking at this code:

```js
        filters: [
            [], // the default value, could also be set via a selector
            {
                [actions.setInitial]: (_, { filters }) => {
                    // called with `actions.setInitial(props.propertyFilters)` in afterMount
                    if (!filters) return []
                    return Object.entries(filters).map(([key, value]) => {
                        let dict = {}
                        dict[key] = value
                    }
                },
                [actions.set]: ...,
                [actions.newFilter]: ...,
                [actions.remove]: ...,
            }
        ]
```

You could simplify it with:

```js
        filters: [
            props.propertyFilters 
                ? Object.entries(props.propertyFilters).map(([key, value]) => ({ [key]: value }) 
                : [],
            {
                [actions.set]: ...,
                [actions.newFilter]: ...,
                [actions.remove]: ...,
            }
        ]
```

Reply:
Done